src/request/mod.rs

```diff
@@ -254,9 +254,10 @@ impl<B: Body> PreparedRequest<B> {
                 .headers()
                 .get(http::header::LOCATION)
                 .ok_or(InvalidResponseKind::LocationHeader)?;
-            let location = location.to_str().map_err(|_| InvalidResponseKind::LocationHeader)?;
 
-            url = self.base_redirect_url(location, &url)?;
+            let location = String::from_utf8_lossy(location.as_bytes());
```

Comment:
What to you think of trying to do `.to_str()` first and if it fails then try the `from_utf8_lossy`? The result can be put in a `Cow`. Avoids allocation on the happy path.

Reply:
Hum, turns out `from_utf8_lossy` already returns a `Cow`. Nevermind that! Rust is great.